File: JAVASCRIPT/factorial.js
function factorial(num) {
    if (num === 0 || num === 1) {
        return 1;
    } else {
        return num * factorial(num - 1);
    }
}

// Taking input from the user
let number = prompt("Enter a non-negative integer to calculate its factorial:");
number = parseInt(number);

// Checking if the input is a non-negative integer
if (!isNaN(number) && number >= 0) {
    // Calculating and displaying the factorial
    let result = factorial(number);
    console.log(`The factorial of ${number} is: ${result}`);
} else {
    console.log("Invalid input. Please enter a non-negative integer.");
}

File: C/factorial.c
#include <stdio.h>

int main() {
    int num, fact = 1;

    // Input from user
    printf("Enter a number: ");
    scanf("%d", &num);

    // Check for negative input
    if (num < 0) {
        printf("Factorial is not defined for negative numbers.\n");
    } else {
        // Calculate factorial
        for (int i = 1; i <= num; ++i) {
            fact *= i;
        }

        // Print the result
        printf("Factorial of %d = %d\n", num, fact);
    }

    return 0;
}

